components/Dialogs/src/MDCAlertController.m

```diff
@@ -186,6 +189,48 @@ - (void)addAction:(MDCAlertAction *)action {
   [self.view setNeedsLayout];
 }
 
+- (BOOL)mdc_adjustsFontForContentSizeCategory {
+  return _mdc_adjustsFontForContentSizeCategory;
+}
+
+- (void)mdc_setAdjustsFontForContentSizeCategory:(BOOL)adjusts {
+  _mdc_adjustsFontForContentSizeCategory = adjusts;
+
+  // Update any action buttons to match the alert controller.
+  for (MDCButton *actionButton in self.actionButtons) {
+    actionButton.mdc_adjustsFontForContentSizeCategory = adjusts;
+  }
+
+  if (_mdc_adjustsFontForContentSizeCategory) {
+    [self updateFontsForDynamicType];
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(contentSizeCategoryDidChange:)
+                                                 name:UIContentSizeCategoryDidChangeNotification
+                                               object:nil];
+  } else {
+    [[NSNotificationCenter defaultCenter] removeObserver:self
+                                                    name:UIContentSizeCategoryDidChangeNotification
+                                                  object:nil];
+  }
+}
+
+// Handles UIContentSizeCategoryDidChangeNotifications
+- (void)contentSizeCategoryDidChange:(NSNotification *)notification {
+  [self updateFontsForDynamicType];
+}
+
+// Update the fonts used based on mdc_preferredFontForMaterialTextStyle and recalculate the
+// preferred content size.
+- (void)updateFontsForDynamicType {
+  self.titleLabel.font = [UIFont mdc_preferredFontForMaterialTextStyle:MDCFontTextStyleTitle];
```

Comment:
Note that there are two similar methods for mdc_preffered*ForMaterialTextStyle.  One is on UIFont, the other is on UIFontDescriptor.

mdf_adjustsFontForContentSizeCategory is locked to system font.
This matches Apple's behavior for adjustsFontForContentSizeCategory.

UIFont mdc_preferredFontForMaterialTextStyle invoked the UIFontDescriptor methods then requests a font using that font descriptor.  The font descriptor in question does not have a font name / family specified which returns system font with the requested attributes (size, weight, etc.).

If clients wish to utilize a custom font, they can invoke UIFontDescriptor mdc_preferredFontDescriptorForMaterialTextStyle and use that descriptor as the basis to request a font with their custom font name / family.  This is likely the hook we would use if we choose to pull into RobotoLoader.  Happy to sit down and work through the details.